internal/core/config/config.go

```diff
@@ -125,6 +129,20 @@ func (c *Config) Save() error {
 
 // KeyStore returns the config keystore.
 func (c *Config) KeyStore() *keystore.KeyStore {
-	path := filepath.Join(c.rootPath, keystoreDir)
+	var path string
+
+	// set timestamp once per process
+	if timestamp == "" {
+		timestamp = fmt.Sprintf("%v", time.Now().UnixNano())
+	}
+
+	// use temporary keystore when VALIST_SIGNER is set
```

Comment:
I think this could be moved into the Load method.  We could store the keystone path as a private config variable and avoid the global variable.

Reply:
awesome suggestion. I added a keystorePath private variable to config, setting the default to `filepath.Join(rootPath, rootDir, keystoreDir)` and overwriting it when VALIST_SIGNER is set like:

```go
if os.Getenv("VALIST_SIGNER") != "" && c.keystorePath == filepath.Join(c.rootPath, keystoreDir) {
    c.keystorePath = filepath.Join(os.TempDir(), keystoreDir, fmt.Sprintf("%v", time.Now().UnixNano()))
}
```